commons/zenoh-shm/src/watchdog/periodic_task.rs

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use thread_priority::ThreadBuilder;
+#[cfg(unix)]
+use thread_priority::{
+    set_current_thread_priority, RealtimeThreadSchedulePolicy, ThreadPriority, ThreadPriorityValue, ThreadSchedulePolicy::Realtime
+};
+
+pub struct PeriodicTask {
+    running: Arc<AtomicBool>,
+}
+
+impl Drop for PeriodicTask {
+    fn drop(&mut self) {
+        self.running.store(false, Ordering::Relaxed)
+    }
+}
+
+impl PeriodicTask {
+    pub fn new<F>(name: String, interval: Duration, mut f: F) -> Self
+    where
+        F: FnMut() + Send + 'static,
+    {
+        let running = Arc::new(AtomicBool::new(true));
+
+        let c_running = running.clone();
+        
+        #[cfg(unix)]
+        let builder = ThreadBuilder::default()
+            .name(name)
+            .policy(Realtime(RealtimeThreadSchedulePolicy::Fifo))
+            .priority(ThreadPriority::Min);
+
+        // todo: deal with windows realtime scheduling
+        #[cfg(windows)]
+        let builder = ThreadBuilder::default().name(name);
+                
+        let _ = builder.spawn(move |result| {
+                if let Err(e) = result {
+                    #[cfg(windows)]
+                    tracing::warn!("{:?}: error setting scheduling priority for thread: {:?}, will run with the default one...", std::thread::current().name(), e);
+                    #[cfg(unix)]
+                    {
+                        tracing::warn!("{:?}: error setting realtime FIFO scheduling policy for thread: {:?}, will run with the default one...", std::thread::current().name(), e);
+                        for priotity in (ThreadPriorityValue::MIN..ThreadPriorityValue::MAX).rev() {
+                            if let Ok(p) = priotity.try_into() {
+                                if set_current_thread_priority(ThreadPriority::Crossplatform(p)).is_ok() {
+                                    tracing::warn!("{:?}: will use priority {}", std::thread::current().name(), priotity);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                //todo: need mlock here!
```

Comment:
```suggestion
                // TODO: need mlock here!
```
In Zenoh we usually use the capital letter version of todo. Just some consistency to better find it in the code.